### PR TITLE
feat: Add automatic-translation extension - EXO-65429

### PR DESCRIPTION
**Pull request by hakermi**
This PR adds a new automatic extension point

### CODE REVIEW

File: webapps/src/main/webapp/javascript/automatic-translation/extensions.js [hunk 1]
```diff
@@ -100,6 +100,17 @@
...
+      id: 'auto-translate',
+      rank: 1000,
+      isEnabled: () => true,
+      labelKey: 'UIActivity.label.translate',
+      click: (message, callback) => {
+        fetchTranslation(message).then(translated => {
+          callback(translated.translation);
+        });
+      },
+    });
```
Comment (rdenarie) [inline webapps/src/main/webapp/javascript/automatic-translation/extensions.js]: why we add this extension ?

Comment (hakermi) [reply webapps/src/main/webapp/javascript/automatic-translation/extensions.js]: > why we add this extension ?

for the translate button in notes and news ?

Comment (rdenarie) [reply webapps/src/main/webapp/javascript/automatic-translation/extensions.js]: Ok, the name of the extension is disturbing. 
You named it 'automatic-translation', but should it be something like 'notes-translation' and 'news-translation' ?
For activities, the xtension is named "ActivityContent" or ('activity', 'comment-action').

Comment (rdenarie) [reply webapps/src/main/webapp/javascript/automatic-translation/extensions.js]: WDYT ?

Comment (hakermi) [reply webapps/src/main/webapp/javascript/automatic-translation/extensions.js]: Ok, but i wanted to have one extension for both. that's why. 
And i really don't know which is better: define one extension or two extensions(news-extension and notes-extension).

Comment (rdenarie) [reply webapps/src/main/webapp/javascript/automatic-translation/extensions.js]: By using only one extension point, you will not be able in the future to add something in the news extension point but not in the notes extension point.
For me, it is better to have 2 extensions point, with a different name

Comment (hakermi) [reply webapps/src/main/webapp/javascript/automatic-translation/extensions.js]: Ok i will make two extensions

File: webapps/src/main/webapp/javascript/automatic-translation/extensions.js [hunk 2]
```diff
@@ -100,6 +100,17 @@
       },
     });
 
+    extensionRegistry.registerExtension('notes-automatic-translation', 'auto-translate', {
```
Comment (rdenarie) [inline webapps/src/main/webapp/javascript/automatic-translation/extensions.js]: you named the extension point "notes-automatic-translation". Does that mean we only add autotranslation extension into ? 
In fact, you named the extension point with a name which is related to the content of the extension, not of what is the extension point.

As the extension point is in notes, translation menu, I would name it "notes", with type 'translation-menu-extension'

Comment (hakermi) [reply webapps/src/main/webapp/javascript/automatic-translation/extensions.js]: name of extension updated

**Comment by exo-swf** [comment]
Your PR triggers too many exo-ci builds! Please finish your work and then, set your PR ready! Thank you

**Review by rdenarie — COMMENTED** [review]
Ok for me. I let @mkrout validates the PR as he have the complete view on the feature

**Review by mkrout — APPROVED** [review]

**Pull request merged**